### PR TITLE
Experience downloading improvements

### PR DESCRIPTION
**Pull request by KirillSto**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: core/gallery/gallery.go [hunk 1]
```diff
@@ -70,6 +70,20 @@ func DownloadManifest(url string) (info *pb.RpcDownloadManifestResponseManifestI
 	return info, nil
 }
 
+func IsInWhitelist(url string) bool {
+	if len(whitelist) == 0 {
+		return true
+	}
+	// nolint:errcheck
+	parsedURL, _ := uri.ParseURI(url)
```
Comment (deff7) [inline core/gallery/gallery.go:78]: don't ignore error here, if url is parsed with error then it definitely invalid

Comment (KirillSto) [reply core/gallery/gallery.go]: Fixed

File: util/builtinobjects/builtinobjects.go [hunk 2]
```diff
@@ -425,8 +425,8 @@ func (b *builtinObjects) downloadZipToFile(url string, progress process.Progress
 	if err = uri.ValidateURI(url); err != nil {
 		return "", fmt.Errorf("provided URL is not valid: %w", err)
 	}
-	if isInWhitelist, err := gallery.IsInWhitelist(url); !isInWhitelist {
-		return "", err
+	if !gallery.IsInWhitelist(url) {
+		return "", fmt.Errorf("URL '%s' is not in whitelist", url)
```
Comment (deff7) [inline util/builtinobjects/builtinobjects.go:429]: probability of sensitive data leak (to logger)

Comment (KirillSto) [reply util/builtinobjects/builtinobjects.go]: Removed URL from message

**Pull request merged**